src/index.ts

```diff
@@ -7,31 +7,39 @@ import type {
 import { HeaderMap } from '@apollo/server';
 import type { WithRequired } from '@apollo/utils.withrequired';
 import type {
+  ALBEvent,
+  ALBResult,
   APIGatewayProxyEvent,
-  APIGatewayProxyEventHeaders,
-  APIGatewayProxyEventQueryStringParameters,
   APIGatewayProxyEventV2,
   APIGatewayProxyResult,
   APIGatewayProxyStructuredResultV2,
   Context,
   Handler,
 } from 'aws-lambda';
 
-export type GatewayEvent = APIGatewayProxyEvent | APIGatewayProxyEventV2;
+export type IncomingEvent =
```

Comment:
This is technically a breaking change, which would be a major version bump and not just a minor. I'd recommend we continue exporting `GatewayEvent` as an alias of `IncomingEvent` and mark it as `@deprecated` with ts-doc and note that it'll be removed in the next major version.